compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -1072,7 +1072,13 @@ extern "C" {
     pub fn LLVMGetPoison(Ty: &Type) -> &Value;
 
     // Operations on metadata
+    // FIXME: deprecated, replace with LLVMMDStringInContext2
     pub fn LLVMMDStringInContext(C: &Context, Str: *const c_char, SLen: c_uint) -> &Value;
+
+    // LLVMMDStringInContext but don't own string
+    pub fn LLVMMDStringInContext2(C: &Context, Str: *const c_char, SLen: size_t) -> &Metadata;
```

Comment:
I can probably replace `LLVMMDStringInContext` with `LLVMMDStringInContext2` all around, using value<->metadata wrappers around, but don't know is it worth it.

Reply:
It looks like some of them are sensible to replace at least. E.g. in consts.rs LLVMMDStringInContext and LLVMMDNodeInContext can be replaced with the 2 variants (just need to replace both of them together and it works out fine).

Reply:
Yes, but `void LLVMAddNamedMetadataOperand(LLVMModuleRef M, const char *Name, LLVMValueRef Val)` 
https://github.com/rust-lang/rust/blob/90a9f69c80812c8694959c1f2d5c336e3300d1e2/compiler/rustc_codegen_llvm/src/consts.rs#L507-L512
https://github.com/llvm/llvm-project/blame/02aa966c135172885182ddd6f5d76883c2402bed/llvm/lib/IR/Core.cpp#L1292-L1300
feeds in LLVMValueRef , but LLVMMDNodeInContext2 will return LLVMModuleRef, so wrap\unwrap (or rewrite LLVMAddNamedMetadataOperand).

Reply:
Also `void LLVMSetMetadata(LLVMValueRef Inst, unsigned KindID, LLVMValueRef Val)` prevents refactor too, sadly.